java/vector/src/main/java/org/apache/arrow/vector/ipc/message/MessageResultData.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector.ipc.message;
+
+import org.apache.arrow.flatbuf.Message;
+
+/**
+ * Class to wrap a MessageResult adding the message body data as an object of type T.
+ *
+ * @param <T> Type parameter for an object that holds message body data
+ */
+public class MessageResultData<T> {
```

Comment:
Why generic? I think a record batch message should always be an ArrowBuf.

Reply:
For my use case, I am just inspecting the message type and don't care about processing any data. I also can not store the data off heap, so any usage of an ArrowBuf would then need to immediately copy it again, which isn't ideal. This is part of a constraint with working within Spark and probably not a common case. I agree that most cases would probably use an ArrowBuf, so would it be preferable to make this non-generic and include an ArrowBuf? As long as there is a way to override it for my needs, I'm fine with that.